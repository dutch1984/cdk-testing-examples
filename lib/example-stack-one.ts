import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as s3 from 'aws-cdk-lib/aws-s3'
import { NagSuppressions } from 'cdk-nag';


export class ExampleStackOne extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const bucket = new s3.Bucket(this, 'MyExampleBucket', {
      versioned: true,
      encryption: s3.BucketEncryption.S3_MANAGED,
      enforceSSL: true
    });

    cdk.Tags.of(bucket).add('application', 'test')
    
    NagSuppressions.addResourceSuppressions(
      bucket,
      [
        {
          id: 'AwsSolutions-S1',
          reason: "A reason for suppressing the warning.",
        }
      ],
      true
    );

  }
}
