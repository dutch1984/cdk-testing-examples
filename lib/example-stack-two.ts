import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as s3 from 'aws-cdk-lib/aws-s3'
import { NagSuppressions } from 'cdk-nag';

export class ExampleStackTwo extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const bucket = new s3.Bucket(this, 'MyExampleBucket', {
      versioned: true
    });
    
    NagSuppressions.addResourceSuppressions(
      bucket,
      [
        {
          id: 'AwsSolutions-S1',
          reason: "A reason for suppressing the warning.",
        },
        {
          id: 'AwsSolutions-S10',
          reason: "A reason for suppressing the warning.",
        }
      ],
      true
    );

  }
}
