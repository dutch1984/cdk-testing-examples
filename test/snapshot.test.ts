import * as cdk from 'aws-cdk-lib';
import { ExampleStackOne } from '../lib/example-stack-one'
import { ExampleStackTwo } from '../lib/example-stack-two';

describe('Snapshot test', () => {
    let stackName: string
    let app: cdk.App

    beforeEach(() => {
        stackName = 'test-example-stack'
        app = new cdk.App()
    });

    test('ExampleStackOne snapshot test', () => {
        // Arrange
        new ExampleStackOne(app, stackName)
        // Act
        const template = app.synth({ force: true }).getStackByName(stackName).template;
        // Assert
        expect(template).toMatchSnapshot()
    });

    test('ExampleStackTwo snapshot test', () => {
        // Arrange
        new ExampleStackTwo(app, stackName)
        // Act
        const template = app.synth({ force: true }).getStackByName(stackName).template;
        // Assert
        expect(template).toMatchSnapshot()
    });
});
    