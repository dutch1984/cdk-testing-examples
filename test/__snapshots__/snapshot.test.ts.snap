// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test ExampleStackOne snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "MyExampleBucket927A78A6": {
      "DeletionPolicy": "Retain",
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-S1",
              "reason": "A reason for suppressing the warning.",
            },
          ],
        },
      },
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "Tags": [
          {
            "Key": "application",
            "Value": "test",
          },
        ],
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "MyExampleBucketPolicyA00A34B4": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-S1",
              "reason": "A reason for suppressing the warning.",
            },
          ],
        },
      },
      "Properties": {
        "Bucket": {
          "Ref": "MyExampleBucket927A78A6",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "MyExampleBucket927A78A6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "MyExampleBucket927A78A6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`Snapshot test ExampleStackTwo snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "MyExampleBucket927A78A6": {
      "DeletionPolicy": "Retain",
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-S1",
              "reason": "A reason for suppressing the warning.",
            },
            {
              "id": "AwsSolutions-S10",
              "reason": "A reason for suppressing the warning.",
            },
          ],
        },
      },
      "Properties": {
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
